### ISSUE TITLE
mono msbuild on Mac/Linux can't find and load libgit2 native binary

### ISSUE THREAD

**Issue by qmfrederik** [AArnott/Nerdbank.GitVersioning#118]
The following project file:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard1.4</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
   <PackageReference Include="Nerdbank.GitVersioning" Version="1.6.25">
       <PrivateAssets>All</PrivateAssets>
   </PackageReference>
  </ItemGroup>
</Project>
```

causes Visual Studio for Mac to crash completely, and msbuild on Mac to fail with the following error:

```
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: git2-1196807
  at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods:git_libgit2_init ()
  at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <48079e5da5364f5db01fd1de5fb12b0d>:0 
  at LibGit2Sharp.Core.NativeMethods..cctor () [0x00054] in <48079e5da5364f5db01fd1de5fb12b0d>:0 
   --- End of inner exception stack trace ---
  at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject.Finalize () [0x00000] in <48079e5da5364f5db01fd1de5fb12b0d>:0 
```

`dotnet build` on the other hand runs just fine.

**Comment by qmfrederik** [comment]
As a temporary workaround, this seems to be a way to prevent that task from running in the context of Mono's msbuild on Mac (which Visual Studio for Mac uses), but still enables it for any other environment:

````
 <ItemGroup Condition="$(MSBuildToolsPath.StartsWith('/Library/Frameworks/Mono.framework/'))">
    <PackageReference Include="Nerdbank.GitVersioning" Version="1.6.25">
        <PrivateAssets>All</PrivateAssets>
    </PackageReference>
  </ItemGroup
```

**Comment by AArnott** [comment]
Thanks for the report and workaround. You mention "MSBuild" but do you really mean *xbuild* ? You clarified once with "Mono's MSBuild" which I don't think exists except as "xbuild".
And more to the point, does this repro with `dotnet build` on mac or only xbuild?

**Comment by qmfrederik** [comment]
@AArnott Nope, msbuild became open source and cross platform a while ago; since then, msbuild is replacing xbuild in the Mono ecosystem.

Mono ships with msbuild 15.2 (which is hosted by mono); .NET Core uses msbuild 15.1 (which hosted by coreclr)

Here's the relevant output:

Mono's msbuild: 
```
Frederiks-MacBook-Pro:~ fcarlier$ which msbuild
/Library/Frameworks/Mono.framework/Versions/Current/Commands/msbuild
Frederiks-MacBook-Pro:~ fcarlier$ msbuild /version
Microsoft (R) Build Engine version 15.2.0.0 (xplat-2017-02/4470478 Fri Apr  7 11:25:14 EDT 2017)
```

.NET Core's msbuild
```
Frederiks-MacBook-Pro:~ fcarlier$ dotnet msbuild /version
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.
```

xbuild's deprecation warning:
```
15.1.548.43366Frederiks-MacBook-Pro:~ fcarlier$ xbuild

>>>> xbuild tool is deprecated and will be removed in future updates, use msbuild instead <<<<
```

**Comment by AArnott** [comment]
I need to spin up my Mac Mini to repro this, I guess, and figure out where my code that helps find the native module is failing.

**Comment by AArnott** [comment]
This reproduces on Linux as well. In short `dotnet build` works fine and `MSBuild 15.1` in Linux and mac when run on mono doesn't work. It's likely because mono doesn't know how to find and load native binaries the way CoreCLR does.

**Comment by AArnott** [comment]
I've figured out the problem and am working on a fix.

**Comment by qmfrederik** [comment]
@aarnott Thanks!

**Comment by AArnott** [comment]
You're welcome. The fix is in 1.6.30.

**Comment by AArnott** [comment]
#123 might still block the scenario, sadly.
